backend/CircleForms/ModelLayer/Publish/PublishService.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Threading.Tasks;
+using CircleForms.Contracts.ContractModels.Response;
+using CircleForms.Database.Models.Posts;
+using CircleForms.Database.Models.Posts.Enums;
+using CircleForms.Database.Services.Abstract;
+using MapsterMapper;
+
+namespace CircleForms.ModelLayer.Publish;
+
+public class PublishService : IPublishService
+{
+    private readonly ICacheRepository _cache;
+    private readonly IMapper _mapper;
+    private readonly IPostRepository _postRepository;
+
+    public PublishService(IMapper mapper, IPostRepository postRepository, ICacheRepository cache)

```

Comment:
Don't think publishing should be in a separate service - it's logically a part of post logic and operate only within posts domain

Reply:
It can scale further, and it should be separate service

Reply:
I represent publishing as a concern, it is no good to mix concerns in one service